examples/mutators/search/README.md

```diff
@@ -0,0 +1,167 @@
+# search: Examples
```

Comment:
@mengqiy do we need the directory organization around `mutators` and `validators`? I find it doesn't add much value and other artifacts (code and tests) do not follow this organization.

Reply:
IMO having `mutators` and `validators` categories on the website can quickly give users an idea what a function is intended to be used.
We currently have 3 categories: `mutators`, `validators` and `runtime` (maybe we can call it flexible).
I classify `starlark` function in the `runtime` category for now. An alternative is that we add it to **both** `mutators` and `validators` categories.

There are at least 2 ways to provide the website builder the category information for a function:
- Maintain 3 directories in the repo: `mutators`, `validators` and `flexible`. Functions in `flexible` will appear in both mutators and validators categories on the website.
- Each function must have a `category.txt` file to indicate if it's mutator, validator or both.

>I think a deeper question is: Is there a programmatic way to identify whether a function is a validator or a mutator?

No, there is no easy way to achieve it.
A validator never mutates resources, but a mutator can choose to behave like a validator if the input is not what it expected.

Reply:
We should not block this PR, let's file an issue to follow up here.

Two orthogonal discussions:

1. Whether and what categories we need.
2. How such category metadata is provided.
3. How the categories are rendered in the catalog.

Currently 2 is done by virtue of directory structure for examples packages? This seems adhoc to me. If the function developer needs to tell us this, then it could be similar to how `usage` is provided (i.e. programmatically instead of some out-of-band convention like this).

Reply:
Filed https://github.com/GoogleContainerTools/kpt/issues/1598